third_party/txt/src/txt/paragraph.cc

```diff
@@ -812,8 +889,16 @@ void Paragraph::Layout(double width, bool force) {
             std::move(code_unit_positions),
             Range<size_t>(run.start(), run.end()),
             Range<double>(glyph_positions.front().x_pos.start,
-                          glyph_positions.back().x_pos.end),
-            line_number, metrics, run.direction());
+                          run.is_placeholder_run()
+                              ? glyph_positions.back().x_pos.start +
+                                    run.placeholder_run()->width
+                              : glyph_positions.back().x_pos.end),
+            line_number, metrics, run.direction(), run.placeholder_run());
+        if (run.is_placeholder_run()) {
+          line_inline_placeholder_code_unit_runs.insert(
```

Comment:
This can be simplified as:
`line_inline_placeholder_code_unit_runs.push_back(line_code_unit_runs.back())`